specification/resourcegraph/resource-manager/Microsoft.ResourceGraph/preview/2018-09-01-preview/graphquery.json

```diff
@@ -0,0 +1,535 @@
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.ResourceGraph/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "description": "Lists all of the available REST API operations.",
+        "operationId": "Operations_List",
+        "x-ms-examples": {
+          "OperationsList": {
+            "$ref": "./examples/OperationsList.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource name. This is GUID value. The display name should be assigned within properties field."
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "eTag": {
+          "type": "string",
+          "readOnly": true,
+          "description": "This will be used to handle Optimistic Concurrency. If not present, it will always overwrite the existing resource without checking conflict."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "x-ms-azure-resource": true,
+      "description": "An azure resource object"
+    },
+    "GraphQueryListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/GraphQueryResource"
+          },
+          "description": "An array of graph queries."
+        }
+      },
+      "description": "Graph query list result."
+    },
+    "GraphQueryResource": {
+      "description": "Graph Query entity definition.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Metadata describing a graph query for an Azure resource.",
+          "$ref": "#/definitions/GraphQueryProperties"
+        }
+      }
+    },
+    "GraphQueryProperties": {
+      "description": "Properties that contain a graph query.",
+      "required": [
+        "query"
+      ],
+      "properties": {
+        "timeModified": {
```

Comment:
>timeModified [](start = 9, length = 12)

Should this be a string or DateTime in the C# SDK? If it's the latter, you should add "format": "datetime"?

Reply:
Btw in general, my suggestion would be to run autorest to generate one of the SDKs (c#, for example) and verify that what it generated seems reasonable.

---
In reply to: [278378779](https://github.com/Azure/azure-rest-api-specs/pull/5478#discussion_r278378779) [](ancestors = 278378779)